lib/nuget-parser/dotnet-core-parser.js

```diff
@@ -123,38 +106,68 @@ function getTargetObjToRun(manifest) {
   return manifest.targets[selectedTargetKey];
 }
 
-
-function parse(fileContent, packageTree) {
-  const manifest = JSON.parse(fileContent);
-
+function validateManifest(manifest) {
   if (!manifest.project) {
     throw new Error('Project field was not found in project.assets.json');
   }
 
-  if (manifest.project.version) {
-    packageTree.version = manifest.project.version;
+  if (!manifest.project.frameworks) {
+    throw new Error('No frameworks were found in project.assets.json');
   }
 
-  const selectedFrameworkObj = getFrameworkObjToRun(manifest);
-  const selectedTargetObj = getTargetObjToRun(manifest);
-
-  initFreqDepsDict();
-
-  const directDependencies = collectFlatList(selectedFrameworkObj.dependencies);
-  debug(`directDependencies: '${directDependencies}'`);
+  if (_.isEmpty(manifest.project.frameworks)) {
+    throw new Error('0 frameworks were found in project.assets.json');
+  }
 
-  directDependencies.forEach(function (directDep) {
-    debug(`First order dep: '${directDep}'`);
-    buildTreeRecursive(selectedTargetObj, directDep, packageTree, 0);
-  });
+  if (!manifest.targets) {
+    throw new Error('No targets were found in project.assets.json');
+  }
 
-  if (!_.isEmpty(freqDeps.dependencies.dependencies)) {
-    packageTree.dependencies['freqSystemDependencies'] = freqDeps.dependencies;
+  if (_.isEmpty(manifest.targets)) {
+    throw new Error('0 targets were found in project.assets.json');
   }
-  // to disconnect the object references inside the tree
-  // JSON parse/stringify is used
-  let pathedTree = JSON.parse(JSON.stringify(packageTree.dependencies));
-  packageTree.dependencies = pathedTree;
 }
 
-module.exports = parse;
+module.exports = {
+  parse: function (tree, fileContent, fileContentParser) {
+    return new Promise(function parseFileContents(resolve, reject) {
+      debug('Trying to parse dot-net-cli manifest');
+      const manifest = fileContentParser.parse(fileContent);
```

Comment:
Looks weird. We are passing `fileContentParser` to this function only to call it once and parse `fileContent`.
Why can't we do `fileContentParser.parse(fileContent)` outside and pass `manifest` as argument?